open-company/open-company-digest
# Switch to only daily digest. TZ aware.

Card: https://trello.com/c/JcvNr1Ew

This is the master PR for:

Auth - https://github.com/open-company/open-company-auth/pull/74

Web - https://github.com/open-company/open-company-web/pull/768

Email - https://github.com/open-company/open-company-email/pull/39


To test email changes:
- Login
- Create a new post
- Send yourself an email digest using the digest service test URL `/_/email/run`
- [x] did you get it by email? Good
- [x] do you NOT see the digest footer about changing settings? Good

To test Web/Auth changes:
- change all 3 of your notifications preferences
- [x] do they "stick" when you reload the app and check? No console or HTTP request errors?
- Send yourself a notification (ask another dev for help or use a 2nd user)
- [x] did you get it with the right medium? Good. Try it with the other 2 mediums too.
- [x] repeat for daily digest

Digest changes to no longer have the concept of daily/weekly, and to have a scheduled run ever hour at the top of the hour, and to send the digest (if one is needed) at 7AM local to the user's TZ.

To test Digest changes:
- Review the added unit tests
- [x] do they seem to be testing the new TZ handling well?
- Create a new post on staging
- Find the TZ that has 7AM coming up nearest to now, whenever now is https://timezonedb.com/time-zones
- Set your TZ to that timezone
- Set your digest to email or Slack
- [x]  Do you get your digest by email or Slack (whichever you picked) at the next top of the hour?
